solidity/contracts/bridge/Wallets.sol

```diff
@@ -517,4 +531,66 @@ library Wallets {
 
         self.registry.closeWallet(wallet.ecdsaWalletID);
     }
+
+    // TODO: Documentation
+    function notifyFundsMoved(
+        Data storage self,
+        bytes20 walletPubKeyHash,
+        bytes32 targetWalletsHash
+    ) external {
+        Wallet storage wallet = self.registeredWallets[walletPubKeyHash];
+        // Check that the wallet is in the MovingFunds state but don't check
+        // if the moving funds timeout is exceeded. That should give a
+        // possibility to move funds in case when timeout was hit but was
+        // not reported yet.
+        require(
+            wallet.state == WalletState.MovingFunds,
+            "ECDSA wallet must be in MovingFunds state"
+        );
+
+        uint32 commitmentSubmittedAt = wallet.movingFundsCommitmentSubmittedAt;
+        require(
+            commitmentSubmittedAt > 0,
+            "Target wallet commitment not submitted yet"
+        );
+        require(
+            /* solhint-disable-next-line not-rely-on-time */
+            block.timestamp >
+                commitmentSubmittedAt +
+                    self.movingFundsCommitmentChallengePeriod,
+            "Target wallet commitment challenge period has not passed yet"
+        );
+
+        // Make sure that the target wallets where funds were moved to are
+        // exactly the same as the ones the source wallet committed to.
+        require(
+            wallet.movingFundsTargetWalletsCommitmentHash == targetWalletsHash,
+            "Target wallets don't correspond to the commitment"
+        );
+
+        // If funds were moved, the wallet has no longer a main UTXO.
+        delete wallet.mainUtxoHash;
+
+        closeWallet(self, walletPubKeyHash);
+    }
+
+    // TODO: Function for committing target wallets for moving funds.
+    //       Can be called only when `movingFundsRequestedAt` field is
+    //       grater than zero, `movingFundsCommitmentSubmittedAt` and
+    //       `movingFundsTargetWalletsCommitmentHash` fields are zeroed, and
+    //       the caller is a wallet's operator. If validation passes, it must
+    //       set the `movingFundsCommitmentSubmittedAt` and
+    //       `movingFundsTargetWalletsCommitmentHash` fields accordingly.
+
+    // TODO: Function for challenging target wallets commitments.
+    //       Can only be called if `block.timestamp` is lesser or equal to
+    //       the sum of `movingFundsCommitmentSubmittedAt` field and
+    //       `movingFundsCommitmentChallengePeriod` governance parameter.
+    //       If the challenge is successful, it must reset the
+    //       `movingFundsCommitmentSubmittedAt` and
+    //       `movingFundsTargetWalletsCommitmentHash` fields to their zero
+    //       value and set `movingFundsRequestedAt` to `block.timestamp`
+    //       to adjust the timeout.
```

Comment:
I know the details are not in the scope of this PR, but maybe we could already mention here what will be checked when validating? I.e.:
- are there "closer" LIVE wallets created before target wallets commitment
- is `wallet_balance` / `no_of_selected_wallets` bigger than `wallet_transfer_max` and, if yes, are there any LIVE wallets created before target wallets commitment was sent that were not chosen?

Reply:
Nope, they are not in the scope of this PR and will be covered in follow-up PRs. I explained the purpose of those TODOs here: https://github.com/keep-network/tbtc-v2/pull/184#discussion_r842434672